cdap-examples/DataCleansing/src/test/java/co/cask/cdap/examples/datacleansing/DataCleansingMapReduceTest.java

```diff
@@ -168,21 +185,23 @@ private void compareData(Long time, String dsName, Set<String> expectedRecords)
 
   private Set<String> getDataFromFile(Long time, String dsName) throws Exception {
     DataSetManager<PartitionedFileSet> cleanRecords = getDataset(dsName);
-    PartitionDetail partition =
-      cleanRecords.get().getPartition(PartitionKey.builder().addLongField("time", time).build());
-
-    if (partition == null) {
-      return Collections.emptySet();
-    }
+    PartitionFilter filter = PartitionFilter.builder().addValueCondition("time", time).build();
+    return getDataFromFilter(cleanRecords.get(), filter);
+  }
 
+  private Set<String> getDataFromFilter(PartitionedFileSet partitionedFileSet,
+                                        PartitionFilter filter) throws IOException {
+    Set<PartitionDetail> partitions = partitionedFileSet.getPartitions(filter);
     Set<String> cleanData = new HashSet<>();
-    Location partitionLocation = partition.getLocation();
-    for (Location location : partitionLocation.list()) {
-      if (location.getName().startsWith("part-")) {
-        try (BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(location.getInputStream()))) {
-          String line;
-          while ((line = bufferedReader.readLine()) != null) {
-            cleanData.add(line);
+    for (PartitionDetail partition : partitions) {
+      Location partitionLocation = partition.getLocation();
+      for (Location location : partitionLocation.list()) {
+        if (location.getName().startsWith("part-")) {
```

Comment:
Since this is a test case for a CDAP application we wrote (DataCleansing), and in that CDAP app's MapReduce, we are in control of the job's configuration, I think it's safe to make this assumption.

It wouldn't be okay to make the assumption if this was code for a generic user-written MapReduce, where the user could override the mapreduce.output.basename key, right?